pkg/storage/mvcc.go

```diff
@@ -3782,18 +3788,28 @@ func MVCCGarbageCollect(
 			iter.Next()
 		}
 
+		// When working with range tombstones, we need to handle the case where last
+		// point key is deleted by range tombstone. In normal case it is handled by
+		// adjusting for metadata above, but for range keys we don't use that because
+		// calculations are different.
+		allPointsDeleted := true
```

Comment:
We should handle this logic in the branch above, which is already responsible for handling the case where we're deleting all versions of a key. The MVCC stats PR that just merged changes `mvccGetMetadata` to return the timestamp where the point key was last changed, which can be used to detect whether all point key versions were removed even if the MVCC range tombstone itself may be above the GC threshold.

Reply:
I started from there, what was happening is that GC is sending requests that are pointing on point key timestamps and that breaks the stats logic. Even if I change gc.Run logic to return timestamp of range key in GCRequest_GCKey instead of first point it would make calling this method unsafe with value under "synth meta" as it may distort the stats. I'll try to see how the logic around meta could be changed to ensure we always adjust stats if we hit first object that is under threshold.